src/settings/panel.tsx

```diff
@@ -0,0 +1,218 @@
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+import { FormComponent, IFormRenderer } from '@jupyterlab/ui-components';
+import { JSONExt } from '@lumino/coreutils';
+import { IChangeEvent } from '@rjsf/core';
+import type { FieldProps } from '@rjsf/utils';
+import validator from '@rjsf/validator-ajv8';
+import { JSONSchema7 } from 'json-schema';
+import React from 'react';
+
+import { IDict, instructions } from './instructions';
+import baseSettings from './schemas/base.json';
+import ProviderSettings from './schemas';
+
+const MD_MIME_TYPE = 'text/markdown';
+const STORAGE_NAME = '@jupyterlite/ai:settings';
+const INSTRUCTION_CLASS = 'jp-lite-ai-settings-instruction';
+
+export const aiSettingsRenderer: IFormRenderer = {
+  fieldRenderer: (props: FieldProps) => <AiSettings {...props} />
+};
+
+export interface ISettingsFormStates {
+  schema: JSONSchema7;
+  instruction: HTMLElement | null;
+}
+
+const WrappedFormComponent = (props: any): JSX.Element => {
+  return <FormComponent {...props} validator={validator} />;
+};
+
+export class AiSettings extends React.Component<
+  FieldProps,
+  ISettingsFormStates
+> {
+  static rmRegistry: IRenderMimeRegistry | null = null;
+
+  constructor(props: FieldProps) {
+    super(props);
+    this._settingsRegistry = props.formContext.settings;
+
+    // Initialize the providers schema
+    const providerSchema = JSONExt.deepCopy(baseSettings) as any;
+    providerSchema.properties.provider = {
+      type: 'string',
+      title: 'Provider',
+      description: 'The AI provider to use for chat and completion',
+      default: 'None',
+      enum: Object.keys(ProviderSettings)
+    };
+    this._providerSchema = providerSchema as JSONSchema7;
+
+    // Initialize the settings from saved one
+    this._provider = this.getCurrentProvider();
+    this._currentSettings = this.getSettings();
+
+    // Initialize the schema
+    const schema = this._buildSchema();
+    this.state = { schema, instruction: null };
+
+    this._renderInstruction();
+
+    // Update the setting registry
+    this._settingsRegistry
+      .set('AIprovider', this._currentSettings)
+      .catch(console.error);
+  }
+
+  async _renderInstruction(): Promise<void> {
+    if (!AiSettings.rmRegistry || !instructions[this._provider]) {
+      this.setState({ instruction: null });
+      return;
+    }
+    let mdStr = instructions[this._provider];
+    mdStr = `---\n\n${mdStr}\n\n---`;
+    const renderer = AiSettings.rmRegistry.createRenderer(MD_MIME_TYPE);
+    const model = AiSettings.rmRegistry.createModel({
+      data: { [MD_MIME_TYPE]: mdStr }
+    });
+    await renderer.renderModel(model);
+    this.setState({ instruction: renderer.node });
+  }
+  /**
+   * Get the current provider from the local storage.
+   */
+  getCurrentProvider(): string {
+    const settings = JSON.parse(localStorage.getItem(STORAGE_NAME) || '{}');
```

Comment:
Wondering how this would play with a custom `overrides.json`, in case some deployments would like to pre-configure a model (and some of its options) by default?

Reply:
That would probably overwrite it.
I don't really know how works `overrides.json`. Can we get its default values with [`annotatedDefaults()`](https://github.com/jupyterlab/jupyterlab/blob/1fb48d636f90928c3b4d883d80030f9bc6da6f73/packages/settingregistry/src/tokens.ts#L529) ?
In this case we could simply replace `'{}'` with `this._settingsRegistry.annotatedDefaults()`.

BTW, we should probably rename `this._settingsRegistry` to `this.settings` for consistency.

Reply:
Looks like `annotatedDefaults()` returns the `overrides.json` contents.

In the constructor, we could check if there is a value in `localStorage.getItem(STORAGE_NAME)`.
If not, read the `annotatedDefaults()` and, if the provider is set in the default settings, fill the local storage with its contents and set the default provider as current. If the provider is not set in the default settings we can probably not use these values.